src/common/transformations/src/transformations/resolve_names_collisions.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <algorithm>
+#include <memory>
+#include <numeric>
+
+#include "transformations/resolve_names_collisions.hpp"
+#include "openvino/op/parameter.hpp"
+#include "openvino/op/result.hpp"
+#include "openvino/op/sink.hpp"
+
+bool ov::pass::ResolveNameCollisions::run_on_model(const std::shared_ptr<ov::Model>& model) {
+    // Next containers are used to fix collisions in autogenerated names
+    // The final list of nodes with collisions
+    std::unordered_set<std::string> unique_friendly_names;
+    std::vector<Node*> nodes_with_conflicts;
+    std::unordered_map<std::string, std::list<Node*>> visited_nodes;
+
+    for (const auto& node : model->get_ordered_ops()) {
+        // Detect names collisions only for nodes with autogenerated names
+        const auto& friendly_name = node->get_friendly_name();
+        if (unique_friendly_names.find(friendly_name) == unique_friendly_names.end())
+            unique_friendly_names.insert(friendly_name);
```

Comment:
Looks like we can get rid of `unique_friendly_names` set because `visited_nodes` will finally have the same unique names.

Reply:
Fixed